Choose your own tidymodels adventure

juliasilge opened the pull request:
This PR starts the blog post draft based on our recent discussions of where folks should see their own analyses fitting into the tidymodels ecosystem.

juliasilge commented:
[Draft post is here](https://deploy-preview-513--tidyverse-org.netlify.app/blog/2021/05/choose-tidymodels-adventure/), for folks looking.

hfrick commented on content/blog/choose-tidymodels-adventure/index.Rmd:
```diff
@@ -0,0 +1,81 @@
+
+![Student loan debt by race across time](https://juliasilge.com/blog/student-debt/index_files/figure-html/unnamed-chunk-3-1.png)
+
+We don't believe most people using tidymodels fall into this first category, but we think that the tools we've built for these kinds of straightforward analyses are well-designed and will set you up for statistical and practical success. Some people will not reach for tidymodels as a whole ecosystem or even parsnip, choosing instead to use underlying model functions, like `lm()` instead of `linea ...
```
The last sentence tripped me up a bit. Maybe it could focus more on "other packages that can be useful on their own" rather than what may not a good fit.

mine-cetinkaya-rundel replied:
I wonder if this is also a good place to say something about "start teaching with tidymodels even if you're not teaching advanced modeling/machine learning to pave the way"? Maybe that's a different post but educators are interested in when/how much to teach tidymodels and I think "just teach parsnip" (or more realistically "just teach parsnip, couple simple recipes, and use broom to tidy output") isn't a bad answer for that question for an intro data science course.

hfrick reviewed (commented):
I think the first paragraph could be a bit more in the spirit of "tidymodels packages that can work fairly stand-alone" (rather than having the strong focus on parsnip only) and then the following paragraphs can continue with their switch to connecting different modelling stages in workflows and workflow sets. I think those two paragraphs explain that nicely!

mine-cetinkaya-rundel commented on content/blog/choose-tidymodels-adventure/index.Rmd:
````diff
@@ -0,0 +1,81 @@
+install.packages("tidymodels")
+```
+
+The packages in tidymodels are designed to work together in a unified ecosystem, but they are flexible and modular; you can use tidymodels packages for certain parts of a modeling analysis without committing to it entirely, when appropriate.
````
```suggestion
The packages in tidymodels are designed to work together in a unified ecosystem, but they are flexible and modular; you can use tidymodels packages for certain parts of a modeling analysis without committing to the entire ecosystem entirely, when preferred.
```

mine-cetinkaya-rundel commented on content/blog/choose-tidymodels-adventure/index.Rmd:
````diff
@@ -0,0 +1,81 @@
+library(tidymodels)
+```
+
+We purposefully write code in small, modular packages both to make them easier to maintain and easier to use in production systems. This does mean that it can be challenging for a newcomer to know where their specific problem fits into this ecosystem. We generally recommend a couple of resources for folks just getting started:
````
"where their specific problem fits into" sounds awkward (should it be in instead of into?) but I'm bad with prepositions so I'm not sure

mine-cetinkaya-rundel commented on content/blog/choose-tidymodels-adventure/index.Rmd:
```diff
@@ -0,0 +1,81 @@
+
+A fully featured ecosystem for modeling and machine learning requires interfaces to, well, models, and the tidymodels package that provides those functions and interfaces is [parsnip](https://parsnip.tidymodels.org/). If your modeling adventure involves small data and straightforward data preprocessing (like that provided by [R's model formula](https://www.tmwr.org/base-r.html#formula)), you may  ...
+
+![Student loan debt by race across time](https://juliasilge.com/blog/student-debt/index_files/figure-html/unnamed-chunk-3-1.png)
```
Would be good to use `knitr::include_graphics()` instead to add alt text with `fig.alt`

mine-cetinkaya-rundel commented on content/blog/choose-tidymodels-adventure/index.Rmd:
```diff
@@ -0,0 +1,81 @@
+
+This post takes on a more specific task for a newcomer to tidymodels; we consider several types of real-world modeling analyses and recommend ways to [choose your own adventure](https://en.wikipedia.org/wiki/Choose_Your_Own_Adventure) in the tidymodels ecosystem.
+
+## Starting with the basics: **parsnip**
```
I imagine the goal of this section is not to repeat the entire Get Started article, but I would suggest adding a bit more here, e.g. how about showing the code for the model accompanying the plot, especially since that plot doesn't come from the linked article. This might also depend on who the audience of this post is, but if true newcomers (knows `lm()` but never before used `linear_reg() %>% set_engine("lm")`) are part of the audience then it would be useful to show the two types of code side by side. The point I like to make here generally is that "yes, we wrote more code to fit this simple model, but starting with this pipeline opens the door up for all the other things you can do within the same pipeline".

EmilHvitfeldt replied:
I second this sentiment:

> "yes, we wrote more code to fit this simple model, but starting with this pipeline opens the door up for all the other things you can do within the same pipeline".

mine-cetinkaya-rundel commented on content/blog/choose-tidymodels-adventure/index.Rmd:
```diff
@@ -0,0 +1,81 @@
+
+When you are setting off on a modeling adventure, it might be worth asking what we even mean by the word "model"; it is a word that gets overloaded really quickly! In the tidymodels ecosystem, we carefully incorporate *both* feature engineering (also called data preprocessing) that must be learned from training data *and* a model fit into a modeling workflow that is estimated together. For exampl ...
+
+![PCA preprocessing is part of the modeling process](https://www.tmwr.org/premade/proper-workflow.svg)
```
I think each of the figures could use either a caption or a sentence in the post that directly links to the figure (e.g. "as shown in the figure below..." so it's clear how they relate to the rest of the post).

mine-cetinkaya-rundel commented on content/blog/choose-tidymodels-adventure/index.Rmd:
```diff
@@ -0,0 +1,81 @@
+
+- our book [*Tidy Modeling with R*](https://www.tmwr.org/) to dig deeper
+
+This post takes on a more specific task for a newcomer to tidymodels; we consider several types of real-world modeling analyses and recommend ways to [choose your own adventure](https://en.wikipedia.org/wiki/Choose_Your_Own_Adventure) in the tidymodels ecosystem.
```
Having read through the whole post and then coming back to this -- I would suggest saying here that the post will outline three scenarios and give a sentence description of each so (1) reader gets a sense of what they're in for and (2) someone who is only interested in one or two of them knows where to jump to. Perhaps most importantly, though, it's helpful to know that this is how the tidymodels team does a high-level categorization of "things" you can do with tidymodels: single model, workflow, workdlowset.

EmilHvitfeldt commented:
I like the post! 
But the beginning of the post has an odd flow to it. You have to read through to the 3-4th paragraph to find the motivation for reading the remaining of the article.

From what I read it appears that the first 2 paragraphs introduce tidymodels and explains its design decision. This is then being used as a launchpad for different "paths". I would personally have liked an earlier hook.

Who is the intended audience for this post?

topepo reviewed (changes requested):
It looks good. I think that it might benefit from adding two pieces to the parsnip and workflows sections. 

For parsnip, maybe something like

> Creating parsnips models can be pretty easy too. The package has an [RStudio add-in](https://www.tidyverse.org/blog/2021/03/tidymodels-2021-q1/#choose-parsnip-models-with-an-rstudio-addin) that you can used to create the code for one or more models. 

and for workflows

> The usemodels package can automate the creation of a workflow by using your data to determine a good recipe along with the model of you choice.

juliasilge commented:
Thank you so much to all for your feedback! I incorporated pretty much everything, I believe, especially about the introduction and making the focus of the post more clear.

Along those lines, I removed the content about other packages that can be used in a modular, non-tidymodels way. That could really be its own post and I think it was just confusing here. So now the three sections are more clearly: parsnip, workflows, workflowsets.
